### DISCUSSION TITLE
Support for other languages, e.g. CoffeeScript

### DISCUSSION THREAD

**Discussion by trylks** [nuxt/nuxt#20351] [Ideas]
If I am not mistaken, as of today, it is possible to write:
```
<script setup lang="ts">
...
<script>
```

But not:
```
<script setup lang="coffee">
...
<script>
```
Adding support for other languages compiling to JavaScript would be nice. There are two groups:

1. Some are plain syntactic transformations: CoffeeScript is the most famous, [Squint](https://github.com/squint-cljs/squint) may have potential.
2. Other languages requiring a runtime would be more complex: PureScript, ClojureScript, Scala.js,…

Group 1 might be a "low-hanging fruit." Would it be very difficult to have it in a functional state? I am only getting started with the JavaScript world. So far, I could see that Vite has support for CoffeeScript, and perhaps that is enough, but I could not make it work with the Nuxt 3 magic. I could achieve the preprocessing of CoffeeScript in a similar context (in Svelte) using Svelte-preprocessor*, but the only preprocessors I found for Vue and Nuxt are for CSS (literally searching for "preprocessor").

It is already possible to use the following:

1. CSS preprocessors and alternative languages, like Sass and Stylus,
2. HTML preprocessors, like Pug.
3. Alternatives to JavaScript, like TypeScript.

These three work today. So I guess there is already some kind of hook to plug other preprocessors. One advantage of Nuxt and Vue is how clean the code is. Doubling down on it with Pug, Stylus, and CoffeeScript would be nice to have, IMHO.

\* If I am not mistaken, I have no idea what I am doing at this point, TBH.

**Comment by peterroe** [comment]
This is a good idea, but unlike `pug`, `less` and other languages, `coffeecript` is difficult to implement, and it is not necessary. As it stands, if so, some existing benefits would be lost, such as IDEA hints

**Comment by trylks** [comment]
Most IDEs support multiple languages nowadays. I do not know about IDEA; it will probably do soon if it still has to.

Why is CoffeeScript difficult to implement?

**Reply by localusercamp** [reply]
This is not only about CoffeeScript, this is about all JS dialects. In my opinion this will never be implemented for this reasons:
- Dialects has very tiny userbases.
- There are tons of dialects and all of them has specific rules that requires a lot of work to implement them. For example: how to use composables inside CoffeScript `-> functions` and so on...

I think the team has so much work to do that will bring more value to more people than that
# 🤷

**Reply by trylks** [reply]
I don't think any of the rules of CoffeeScript would require any attention: preprocess the CoffeeScript into JavaScript and use it as if the programmer had written JavaScript in the first place. After preprocessing, you can forget about CoffeeScript. Only if you want to be extra-friendly to CoffeeScript developers; you may translate the error messages to the original code map.

The team would have less work to do if they used better languages.
# 😜
